ballerina/swiftmt_parser.bal

```diff
@@ -0,0 +1,272 @@
+# + return - Returns the customized XML with renamed tags and components. If a transaction sequence is required, it is added to the message. In case of an error, returns an error indicating the issue.
+isolated function customizeGeneratedXml(xml customXml) returns xml|error {
+    boolean isTransactionSequencerequired = false;
+    string messageType = (customXml/**/<messageType>).data();
+    boolean isNotCommonMessageType = check int:fromString(messageType.substring(1, 3)) < 90;
+
+    foreach xml tagElement in (customXml/**/<block3>).elementChildren() {
+        if (tagElement/<name>).data().equalsIgnoreCaseAscii("121") {
+            tagElement.setName("NdToNdTxRef");
+        } else if (tagElement/<name>).data().equalsIgnoreCaseAscii("119") {
+            tagElement.setName("ValidationFlag");
+        }
+    }
+
+    string[][]? csvContent1 = FIELDNAMESPEC[messageType.substring(0, 1)];
+    if csvContent1 is null {
+        return error("SWIFT message type is invalid or not supported.");
+    }
+    foreach xml tagElement in (customXml/**/<block4>).elementChildren() {
+        string name = (tagElement/<name>).data();
+        if name.equalsIgnoreCaseAscii("21") {
+            isTransactionSequencerequired = true;
+        }
+        tagElement.setName("MT".concat(name));
+        string componentName = "";
+        foreach xml component in tagElement.elementChildren() {
+            if component.getName().equalsIgnoreCaseAscii("component") {
+                foreach string[] line in csvContent1 {
+                    if name.equalsIgnoreCaseAscii(line[0]) {
+                        foreach int index in 1 ... line.length() - 1 {
+                            if index.toString() != component.getAttributes()["number"] {
+                                continue;
+                            }
+                            if name.equalsIgnoreCaseAscii("50F") || name.equalsIgnoreCaseAscii("59F") {
+                                if index > 1 {
+                                    if line[index].equalsIgnoreCaseAscii("CdTyp") {
+                                        foreach string[] line2 in IDENTIFYTAG {
+                                            if component.data().equalsIgnoreCaseAscii(line2[0]) {
+                                                componentName = line2[1];
+                                                component.setName("CdTyp");
+                                            }
+                                        }
+                                    } else {
+                                        component.setName(componentName);
+                                    }
+                                } else {
+                                    component.setName(line[index]);
+                                }
+                            } else {
+                                component.setName(line[index]);
+                            }
```

Comment:
Early returns (continues) would reduce nesting.

Reply:
fixed in `99b8222`